src/gui/qgsmanageconnectionsdialog.cpp

```diff
@@ -1194,6 +1260,99 @@ void QgsManageConnectionsDialog::loadDb2Connections( const QDomDocument &doc, co
   }
 }
 
+void QgsManageConnectionsDialog::loadHanaConnections( const QDomDocument &doc, const QStringList &items )
+{
+  QDomElement root = doc.documentElement();
+  if ( root.tagName() != QLatin1String( "qgsHanaConnections" ) )
+  {
+    QMessageBox::information( this,
+                              tr( "Loading Connections" ),
+                              tr( "The file is not a HANA connections exchange file." ) );
+    return;
+  }
+
+  QString connectionName;
+  QgsSettings settings;
+  settings.beginGroup( QStringLiteral( "/HANA/connections" ) );
+  QStringList keys = settings.childGroups();
+  settings.endGroup();
+  QDomElement child = root.firstChildElement();
+  bool prompt = true;
+  bool overwrite = true;
+
+  while ( !child.isNull() )
+  {
+    connectionName = child.attribute( QStringLiteral( "name" ) );
+    if ( !items.contains( connectionName ) )
+    {
+      child = child.nextSiblingElement();
+      continue;
+    }
+
+    // check for duplicates
+    if ( keys.contains( connectionName ) && prompt )
+    {
+      int res = QMessageBox::warning( this,
+                                      tr( "Loading Connections" ),
+                                      tr( "Connection with name '%1' already exists. Overwrite?" )
+                                      .arg( connectionName ),
+                                      QMessageBox::Yes | QMessageBox::YesToAll | QMessageBox::No | QMessageBox::NoToAll | QMessageBox::Cancel );
+      switch ( res )
+      {
+        case QMessageBox::Cancel:
+          return;
+        case QMessageBox::No:
+          child = child.nextSiblingElement();
+          continue;
+        case QMessageBox::Yes:
+          overwrite = true;
+          break;
+        case QMessageBox::YesToAll:
+          prompt = false;
+          overwrite = true;
+          break;
+        case QMessageBox::NoToAll:
+          prompt = false;
+          overwrite = false;
+          break;
+      }
+    }
+
+    if ( keys.contains( connectionName ) && !overwrite )
+    {
+      child = child.nextSiblingElement();
+      continue;
+    }
+
+    //no dups detected or overwrite is allowed
+    settings.beginGroup( "/HANA/connections/" + connectionName );
+
+    settings.setValue( QStringLiteral( "/driver" ), child.attribute( QStringLiteral( "driver" ) ) );
```

Comment:
given the apparent 1-1 matching between settings name and XML name, a string list and a loop over it would make it more obvious and less prone to copy&paste errors

Reply:
I would just add that when the connections API will be implemented there won't be need to duplicate code here or in the data items, the connections API will be used to read/store connections from/to QgsSettings and to create children (schema, tables, whatever HANA supports) for the data items.

Reply:
The Connection API for HANA plugin is implemented in a4819007.

Reply:
In Debug mode, QgsHanaProviderConnection class is not used at all. Should I enable it somewhere or Connection API is not activated yet?

Reply:
The connection API is used internally by the providers in dataitems implementations and externally by processing and DB manager. The current situation is that not all data providers connection APIs are implemented, when that will eventually happen we will be able to remove a lot of custom code and use the connections API abstraction layer in **all** data items implementations and in other parts of the code, DB manager included.

In HANA, you should probably use the connections API for the dataitems and to read/write the connections from/to QgsSettings. If you are planning to implement HANA in DB-manager, that is another good candidate for using the connections API.

Note: debug mode does not affect the usage of the API in any way.